mail-server/networking.nix

```diff
@@ -17,7 +17,7 @@
 { domain, host_prefix, enable_imap, enable_pop3 }:
 
 {
-  hostName = "${host_prefix}.${domain}";
+  #hostName = "${host_prefix}.${domain}";
```

Comment:
why's that?

Reply:
As far as I know, `hostName` doesn't have any functionality other than to give the machine a name for recognition. Since I set this myself in my config (e.g. Home machine is just named "nixos"), I got an error because it was declared multiple times.

Reply:
It is however essential that the `hostName` is set correctly for a mail server, as that is the name `postfix` will use in the banner when initiating a `smtp` connection.

Reply:
Ah I see, I'll have a closer look

Reply:
yeah, same thing like in https://github.com/r-raymond/nixos-mailserver/pull/4
then it should be removed, not commented.

Reply:
Thinking about it, one could maybe work around this problem by specifying the hostname at various places in the config files of `postfix` and `dovecot`, but I'm unsure what the benefits are. It is _very_ common practice to use set the correct FQDN a the host name.

Reply:
For postfix I think [this](http://www.postfix.org/STANDARD_CONFIGURATION_README.html#stand_alone) is it.

Reply:
Admittedly I don't have much experience with mail servers, but setting hostName for this just kinda feels wrong, a server doesn't have to be only for mail which `mail.example.com` would indicate.

Reply:
I was referring to
```
smtpd_banner = $myhostname ESMTP NO UCE
```
which of course can be changed. However I think the domain name leaks into the config in multiple places. To be really sure that nothing is wrong we would have to find all the places and test the server a few weeks, seeing if we end up on any black lists.

To me there is no compelling argument why it is necessary to do that. But if you feel strongly about that, maybe we can work on that in a future release.

Reply:
> but I'm unsure what the benefits are
people tend to name their machines as pets, etc... via `hostName`, so will be disappointed here

Jokes aside, NixOS is very declarative, but still allows hidden interconnections. Examples are `hostName` here (which implicitly defines how `postfix` works) and `systemPackages` (which other services may require implicitly). So, if it is possible to remove these hidden things, better do that, so current modules can server as a source for knowledge, apart from system spec.

Reply:
I should probably read more about how all of this works, I'll just uncomment that line for now.

I'm curious: If I have a mailserver with hostname set to something other than `mail.example.com`, what would not work?

Reply:
> So, if it is possible to remove these hidden things, better do that, so current modules can serve as a source for knowledge, apart from system spec.

That is a good point, but might be a little out of scope for the first version.

Reply:
> I'm curious: If I have a mailserver with hostname set to something other than mail.example.com, what would not work?

Your mail would probably end in every spam filter, if the other smtp server would not simply reject the connection. Note that it is not important what your `FQDN` is, but that your `DNS` record, reverse `DNS` record, `smtp` banner, ... agree.